tools/ping1d-simulation.py

```diff
@@ -39,23 +39,27 @@ def read(self):
                     # we decoded a message from the client
                     self.handleMessage(self.parser.rx_msg)
 
-        except Exception as e:
+        except EnvironmentError as e:
             if e.errno == errno.EAGAIN:
                 pass # waiting for data
             else:
                 print("Error reading data", e)
 
+        except KeyError as e:
+           print("skipping unrecognized message id")
```

Comment:
It would help to print the id and the package content in hex.

Reply:
done